### ISSUE TITLE
Products: Tapping on left edge of editable cell after making changes trigger "Discard changes" prompt

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#2249]
**Describe the bug**

Since the change in https://github.com/woocommerce/woocommerce-ios/pull/2158 I expect that I can tap anywhere in a product cell for an editable field and edit that field.

However, once I make an edit to a product any tap on the left edge of the screen will trigger the back button and "Discard changes" prompt instead of allowing me to edit another field, even if the tap is within a product cell.

That "Discard changes" behavior exists in previous versions, but it now clashes with the expectation that I can tap anywhere in a product cell to edit it. cc @pmusolino 

**To Reproduce**

1. Open a product detail.
2. Open the Price section.
3. Tap over the word "Price" (near the left edge of the cell) and edit the price.
4. Tap over the word "Sale" in "Sale Price" (near the left edge of the cell) to try to edit the sale price.
5. Notice that the "Discard changes" action sheet appears.

**Screenshots**

Screencast: https://cloudup.com/cOkXaz-fP_b

**Mobile Environment**
Please include:
- Device: iPhone XS
- iOS version: 13.4.1
- WooCommerce iOS version: 4.2.0.0 TestFlight beta

**Comment by astralbodies** [comment]
I bet you're triggering the "swipe from left" to go back a screen causing the discard thing to come up. @pmusolino can you confirm?

**Comment by pmusolino** [comment]
I investigated it, going back through release 4.0 and 4.1 (where this problem was introduced), and I confirm that it seems a bug, introduced when I implemented a way to present the discard action sheet also when the user does a swipe back: https://github.com/woocommerce/woocommerce-ios/pull/2146

I have to investigate if there is a way to avoid this behavior, none of us in the testing phase had noticed it. Nice catch @rachelmcr!

**Comment by jaclync** [comment]
yea this "Discard changes" prompt is triggered by the swipe back gesture from my testing:

<img width="589" alt="Screen Shot 2020-05-26 at 10 30 38 AM" src="https://user-images.githubusercontent.com/1945542/82854806-1822a800-9f3c-11ea-9b7a-040af6adbe87.png">

TLDR: this issue might be tricky to fix, since iOS determines what gestures are triggered and it's often not the tap gesture even though we think we are simply tapping on a row.

I was checking what gestures trigger the swipe back callback (the `otherGestureRecognizer` in the [`gestureRecognizer(_:shouldBeRequiredToFailBy:)` callback](https://developer.apple.com/documentation/uikit/uigesturerecognizerdelegate/1624222-gesturerecognizer), and there are all kinds of them 😅 

when I tapped on the sale cell, I've seen:
```
UITapGestureRecognizer
UIScrollViewPanGestureRecognizer
UIScrollViewDelayedTouchesBeganGestureRecognizer
_UISwipeActionPanGestureRecognizer
UILongPressGestureRecognizer
```

when I performed an intentional swipe back, the gesture was:
```
_UISwipeActionPanGestureRecognizer
```

we could potentially block off certain gesture recognizers like `UITapGestureRecognizer` and `UILongPressGestureRecognizer`, but it feels kind of hacky and there could still be many other types of gesture recognizers that we don't cover. @pmusolino what do you think?

**Comment by jaclync** [comment]
It sounds like there isn't a good solution for us to refine the swipe back gesture, since a lot of the gesture handling is within UIKit. Maybe we can close this issue for now?

**Comment by pmusolino** [comment]
@jaclync I agree with you, for the moment it's better to close this issue 👍 closing it now, but feel free to re-open it
